models/userDao.js

```diff
@@ -0,0 +1,82 @@
+const appDataSource = require('./dataSource');
+
+const getUserIdByKakaoId = async(kakaoId) => {
+  return await appDataSource.query(`
+    SELECT
+      id AS userId
+    FROM users
+    WHERE kakao_id = ?;`, [ kakaoId ]
+  );
+}
+
+const storeKakaoUserInfo = async(kakaoUserInfo) => {
+  const kakaoId = kakaoUserInfo.id;
+  const nickname = kakaoUserInfo['kakao_account'].profile.nickname;
+  const profileImageUrl = kakaoUserInfo['kakao_account'].profile['profile_image_url'];
+  const email = kakaoUserInfo['kakao_account'].email;
+  const birthday = kakaoUserInfo['kakao_account'].birthday;
+  let gender = kakaoUserInfo['kakao_account'].gender;
```

Comment:
해당 부분은 service에서 비구조화 할당을 통해 각 데이터를 받아오는 것으로 수정해주세요!

Reply:
`const storeKakaoUserInfo = async(kakaoUserInfo) => {
  
  let { kakaoId, nickname, profileImageUrl, email, birthday, gender, genderId } = {
    kakaoId : kakaoUserInfo.id,
    nickname : kakaoUserInfo['kakao_account'].profile.nickname,
    profileImageUrl : kakaoUserInfo['kakao_account'].profile['profile_image_url'],
    email : kakaoUserInfo['kakao_account'].email,
    birthday : kakaoUserInfo['kakao_account'].birthday,
    gender : kakaoUserInfo['kakao_account'].gender,
    genderId : 0
  }

  if (gender === 'male') {
    genderId = 1;
  } else if (gender === 'female') {
    genderId = 2;
  } else {
    genderId = 3;
  }
  
  const [ userId ] = await userDao.getUserIdByKakaoId(kakaoId);

  if (!userId) {
    await userDao.storeKakaoUserInfo(kakaoId, nickname, profileImageUrl, email, birthday, genderId)
  }
};`

비구조화 할당은 어떻게 적용해야 할지 잘 모르겠습니다...
service 내에서 이런 식으로 수정하면 될까요?

Reply:
은송님! 지금 `kakaoUserInfo`객체에 담기는 `kakao_account`역시 객체로 보여지는데 이 부분도 비구조화 할당을 통해 하위 데이터를 불러올 수 있을 것 같아요!

```suggestion
  const kakaoId = kakaoUserInfo.id 
  const { email, birthday, gender } = kakaoUserInfo.kakao_account;
  const { nickname, profileImageUrl } = kakaoUserInfo.kakao_account.profile;
```

Reply:
오 이런 식으로 작성하면 훨씬 깔끔해지는군요! 감사합니다.
그렇게 수정하도록 하겠습니다!